src/main/java/com/example/report/service/ReportService.java

```
@@ -494,6 +684,133 @@ public List<ReportDTO> getAllReports() {
+
+        for (Benefit benefit : benefits) {
+            table.addCell(String.valueOf(benefit.getId()));
+            table.addCell(benefit.getName());
+
+            if (benefit.getBenefitTypeName().equals(ACCRUAL)){
+                table.addCell(String.valueOf(amounts.get(benefit.getName())));
+                table.addCell("");
+                totalAccruals = totalAccruals.add(amounts.get(benefit.getName()));
+            } else {
+                table.addCell("");
+                table.addCell(String.valueOf(amounts.get(benefit.getName())));
+                totalDeductions = totalDeductions.add(amounts.get(benefit.getName()));
+            }
+        }
+
+        totalDeductions = totalDeductions.add(amounts.get(PENSIONS_FUND));
+        totalDeductions = totalDeductions.add(amounts.get(PERSONAL_INCOME_TAX));
+
+        table.addCell("");
+        table.addCell("Pensions Fund");
+        table.addCell("");
+        table.addCell(amounts.get(PENSIONS_FUND).toString());
+
+        table.addCell("");
+        table.addCell("Personal Income Tax");
+        table.addCell("");
+        table.addCell(amounts.get(PERSONAL_INCOME_TAX).toString());
+
+        table.addCell("");
+        table.addCell("TOTAL");
+        table.addCell(String.valueOf(totalAccruals));
+        table.addCell(String.valueOf(totalDeductions));
+
+        table.addCell("");
+        table.addCell("NET Payable Amount");
+        table.addCell("");
+        table.addCell(String.valueOf(totalAccruals.subtract(totalDeductions)));
+
+        document.add(table);
+    }
+
+
+    public void extractPaySlipInPDF(com.lowagie.text.Document document, Long employeeId, Long reportId) {
+
+        List<ReportEntry> allByEmployeeIdAndReportId = reportEntryRepository.findAllByEmployeeIdAndReportId(employeeId, reportId);
+        Employee employee = allByEmployeeIdAndReportId.get(0).getEmployee();
+        Report report = allByEmployeeIdAndReportId.get(0).getReport();
```

Let's fetch data about Employee and Report from corresponding repository (we should throw exceptions if report or employee are not found)